scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/core/service/IDataFileService.java

```diff
@@ -0,0 +1,35 @@
+package edu.asu.diging.scriptoocloud.core.service;
+
+import edu.asu.diging.scriptoocloud.core.exceptions.DataFileStorageException;
+import edu.asu.diging.scriptoocloud.core.model.impl.DataFile;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+
+/**
+ * The interface associated with user-uploaded DataFiles.  The interface handles storing the
+ * files on the file system (using the FileSystemService) and storing DataFile data in the database.
```

Comment:
The second sentence should go on the implementing class, since the interface by itself doesn't do anything. Here, javadoc should describe how or when to use this interface and what its implementations do (e.g. instances of this class do x). In theory, there could be another implementation that stores files in S3, so that this javadoc would not be correct anymore.